pkg/apis/servicecatalog/v1alpha1/types.go

```diff
@@ -364,7 +364,7 @@ type BindingSpec struct {
 
 	// SecretName is the name of the secret to create in the Binding's
 	// namespace that will hold the credentials associated with the Binding.
-	SecretName string `json:"secretName"`
+	SecretName string `json:"secretName,omitempty"`
```

Comment:
This shouldn't be omitempty, because it's not optional - you just added a default for it.